boltz/musig.go

```diff
@@ -0,0 +1,95 @@
+package boltz
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2/schnorr/musig2"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	liquidtx "github.com/vulpemventures/go-elements/transaction"
+)
+
+type MusigSession struct {
+	*musig2.Session
+	outputs []OutputDetails
+	idx     int
+}
+
+func NewSigningSession(outputs []OutputDetails, idx int) (*MusigSession, error) {
+	tree := outputs[idx].SwapTree
+	ctx, err := musig2.NewContext(
+		tree.ourKey,
+		false,
+		musig2.WithTweakedContext(tree.taprootTweak),
+		musig2.WithKnownSigners([]*secp256k1.PublicKey{tree.boltzKey, tree.ourKey.PubKey()}),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	session, err := ctx.NewSession()
+	if err != nil {
+		return nil, err
+	}
+
+	return &MusigSession{session, outputs, idx}, nil
+}
+
+func (session *MusigSession) Finalize(transaction Transaction, network *Network, boltzSignature *PartialSignature) error {
+	partialSignature, err := hex.DecodeString(boltzSignature.PartialSignature)
+	if err != nil {
+		return err
+	}
+
+	nonce, err := hex.DecodeString(boltzSignature.PubNonce)
+	if err != nil {
+		return err
+	}
+
+	all, err := session.RegisterPubNonce([66]byte(nonce))
```

Comment:
A check to make sure the length of `nonce` is actually 66 before doing the cast would be appropriate